Fix install task and lock file used for yarn

pvliss opened the pull request:
This PR contains the following fixes

1. Switched back to the synthetic tasks, i.e `npm_install` and `yarn_install` as there is no `yarnInstall` task.
2. Fix input file for yarn is `yarn.lock`

Fixes #9451

-   Please make sure the below checklist is followed for Pull Requests.

-   [ ] [Travis tests](https://travis-ci.org/jhipster/generator-jhipster/pull_requests) are green
-   [x] Tests are added where necessary
-   [x] Documentation is added/updated where necessary
-   [x] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/master/CONTRIBUTING.md) are followed

<!--
Please also reference the issue number in a commit message to [automatically close the related Github issue](https://help.github.com/articles/closing-issues-via-commit-messages/)

Note: It is also possible to add `[skip ci]` to your commit message to skip Travis tests
-->

pvliss commented:
@atomfrede + @pascalgrimaud as this issue when unnoticed for some time, I believe it would be nice to also add some CI test for yarn as well. I am thinking the following variations from ngx-default:

1. ngx, h2, maven, yarn, langs: en + fr
2. react, h2, gradle, yarn, langs: en + fr

I know that there are already many tests and hence maybe adding them to the [daily](https://github.com/hipster-labs/jhipster-daily-builds) would be better?
What do you guys think?

atomfrede commented on generators/server/templates/gradle/profile_dev.gradle.ejs:
```diff
@@ -54,7 +58,7 @@ task webpackBuildDev(type: <%= _.upperFirst(clientPackageManager) %>Task) {
     outputs.dir("<%= CLIENT_DIST_DIR %>")
 
-    dependsOn <%= clientPackageManager %>Install
+    dependsOn <%= clientPackageManager %>_install
```
If i get it right, this means the `install` task will be executed regardless if gradle would consider it up to date? As we already differ between `npm` and `yarn` for the lock file we could also put the correct install task to that block I would say.

pvliss replied:
I do not think so. From what I understand as long as inputs + outputs do not change then the whole `webpackBuildDev` task will be skipped due to incremental compilation taking effect. I believe the declaration of `dependsOn npm_install` should be after the declarations of inputs+ outputs in order for everything to work properly but I can check if you like.

atomfrede replied:
Let me give it a try, maybe I am wrong here.

atomfrede replied:
@pvliss We should try to use `npmInstall` instead of `npm_install`:

Second run when using `npmInstall`
```
./gradlew webpackBuildDev                                                                                              1.7m  Wed 20 Mar 2019 05:03:48 PM CET
> Task :nodeSetup SKIPPED
> Task :npmSetup SKIPPED
> Task :npmInstall UP-TO-DATE
> Task :webpackBuildDev UP-TO-DATE

BUILD SUCCESSFUL in 2s
2 actionable tasks: 2 up-to-date
```

Second run when using `npm_install` (as basically `npm_install` does nothave proper up-2-date checking it is executed, but does nothing)

```
./gradlew webpackBuildDev                                                                                           34.8s  Wed 20 Mar 2019 05:04:44 PM CET
> Task :nodeSetup SKIPPED
> Task :npmSetup SKIPPED

> Task :npm_install
npm WARN acorn-dynamic-import@4.0.0 requires a peer of acorn@^6.0.0 but none is installed. You must install peer dependencies yourself.
npm WARN bootstrap@4.3.1 requires a peer of jquery@1.9.1 - 3 but none is installed. You must install peer dependencies yourself.
npm WARN bootstrap@4.3.1 requires a peer of popper.js@^1.14.7 but none is installed. You must install peer dependencies yourself.
npm WARN jest-preset-angular@6.0.2 requires a peer of jest@^23.0.0 but none is installed. You must install peer dependencies yourself.
npm WARN ts-jest@23.1.4 requires a peer of babel-core@^6.0.0 || ^7.0.0-0 but none is installed. You must install peer dependencies yourself.
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@1.2.7 (node_modules/fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.2.7: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"x64"})

up to date in 28.619s

> Task :webpackBuildDev UP-TO-DATE

BUILD SUCCESSFUL in 37s
2 actionable tasks: 1 executed, 1 up-to-date
```

atomfrede replied:
@pvliss We should try to use `npmInstall` instead of `npm_install`:

Second run when using `npmInstall`
```
./gradlew webpackBuildDev                                                                                              1.7m  Wed 20 Mar 2019 05:03:48 PM CET
> Task :nodeSetup SKIPPED
> Task :npmSetup SKIPPED
> Task :npmInstall UP-TO-DATE
> Task :webpackBuildDev UP-TO-DATE

BUILD SUCCESSFUL in 2s
2 actionable tasks: 2 up-to-date
```

Second run when using `npm_install` (as basically `npm_install` does nothave proper up-2-date checking it is executed, but does nothing)

```
./gradlew webpackBuildDev                                                                                           34.8s  Wed 20 Mar 2019 05:04:44 PM CET
> Task :nodeSetup SKIPPED
> Task :npmSetup SKIPPED

> Task :npm_install
npm WARN acorn-dynamic-import@4.0.0 requires a peer of acorn@^6.0.0 but none is installed. You must install peer dependencies yourself.
npm WARN bootstrap@4.3.1 requires a peer of jquery@1.9.1 - 3 but none is installed. You must install peer dependencies yourself.
npm WARN bootstrap@4.3.1 requires a peer of popper.js@^1.14.7 but none is installed. You must install peer dependencies yourself.
npm WARN jest-preset-angular@6.0.2 requires a peer of jest@^23.0.0 but none is installed. You must install peer dependencies yourself.
npm WARN ts-jest@23.1.4 requires a peer of babel-core@^6.0.0 || ^7.0.0-0 but none is installed. You must install peer dependencies yourself.
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@1.2.7 (node_modules/fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.2.7: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"x64"})

up to date in 28.619s

> Task :webpackBuildDev UP-TO-DATE

BUILD SUCCESSFUL in 37s
2 actionable tasks: 1 executed, 1 up-to-date
```

pvliss replied:
Good find. Let me update the PR then.

atomfrede replied:
Sorry for double posting, but the connection currently is really bad.

pvliss replied:
No problem

atomfrede commented:
As yarn is cli flag I would maybe not add it to our regular builds, but to the daily builds. @jhipster/developers Do we have yarn/npm statistics somewhere?

pascalgrimaud commented:
it should go to daily
If you know how to generate a new azure pipeline file, I can merge it and customize the scheduler

Otherwise, I'll do it one day:)

pvliss commented:
Thank you both for your input. I will create an issue at blueprints(so that I do not forget) and will try to implement it or otherwise assign it to you @pascalgrimaud :)

pvliss commented:
Oh I see there is already one: https://github.com/hipster-labs/jhipster-daily-builds/issues/4

atomfrede commented:
@pvliss Do we need to change the `profile_prod.gradle` too?

atomfrede commented:
@pvliss Do we need to change the `profile_prod.gradle` too?

pvliss commented:
@atomfrede Thought I checked but let me check again

pvliss commented:
You are right. We need to change that too. Will do right away

pvliss commented:
@atomfrede So the proper name for the `YarnInstallTask` is simply `yarn`, I have also fixed the prod profile and moved back to proper names instead of synthetic ones. Please give it another look. Thanks

atomfrede commented on generators/server/templates/gradle/profile_dev.gradle.ejs:
```diff
@@ -45,7 +45,11 @@ bootRun {
 
 <%_ if (!skipClient) { _%>
 task webpackBuildDev(type: <%= _.upperFirst(clientPackageManager) %>Task) {
+    <%_ if (clientPackageManager==='npm') { _%>
```
For readability I would but spaces here: `clientPackageManager === 'npm'` (and at all other places too).

pvliss replied:
:+1: Done

pvliss commented:
Thanks @atomfrede
